controllers/envtest/volumegroupcontent_controller_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package envtest
+
+import (
+	"context"
+	"time"
+
+	volumegroupv1 "github.com/IBM/csi-volume-group-operator/api/v1"
+	"github.com/IBM/csi-volume-group-operator/controllers/envtest/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+var _ = Describe("Controller", func() {
+	Describe("Test controllers", func() {
+		Context("Test VGC controllers", func() {
+
+			BeforeEach(func() {
+				err := cleanTestNamespace()
+				Expect(err).ToNot(HaveOccurred())
+			})
+			It("Should not delete vgc when vgclass deletion policy is retain", func(done Done) {
+				By("Creating a volumeGroup resources and set VGClass deletion policy to retain")
+				retainDeletionPolicy := volumegroupv1.VolumeGroupContentRetain
+				err := utils.CreateResourceObject(Secret, k8sClient)
+				Expect(err).NotTo(HaveOccurred())
+
+				err = utils.CreateResourceObject(VGClass, k8sClient)
+				Expect(err).NotTo(HaveOccurred())
+				vgclass := &volumegroupv1.VolumeGroupClass{}
+				err = utils.GetNamespacedResourceObject(VGClassName, Namespace, vgclass, k8sClient)
+				Expect(err).NotTo(HaveOccurred())
+				vgclass.VolumeGroupDeletionPolicy = &retainDeletionPolicy
+				err = k8sClient.Update(context.TODO(), vgclass)
+				Expect(err).NotTo(HaveOccurred())
```

Comment:
can add arg to create create resource method

Reply:
done